torch/distributed/elastic/multiprocessing/api.py

```diff
@@ -470,11 +471,12 @@ def __init__(
 
     def start(self) -> None:
         """Start processes using parameters defined in the constructor."""
-        signal.signal(signal.SIGTERM, _terminate_process_handler)
-        signal.signal(signal.SIGINT, _terminate_process_handler)
-        if not IS_WINDOWS:
-            signal.signal(signal.SIGHUP, _terminate_process_handler)
-            signal.signal(signal.SIGQUIT, _terminate_process_handler)
+        if threading.current_thread() is threading.main_thread():
+            signal.signal(signal.SIGTERM, _terminate_process_handler)
+            signal.signal(signal.SIGINT, _terminate_process_handler)
+            if not IS_WINDOWS:
+                signal.signal(signal.SIGHUP, _terminate_process_handler)
+                signal.signal(signal.SIGQUIT, _terminate_process_handler)
```

Comment:
We should also add some tests for this behavior

Reply:
Would be nice to add a test to verify elastic can work on a child thread to avoid regressions

There's some similar tests in https://github.com/pytorch/pytorch/blob/ed06772e350c5d550e6a2212d01f5379db475bf3/test/distributed/elastic/multiprocessing/api_test.py#L350-L363 which you can base it off of